script.rb

```diff
@@ -13,43 +13,46 @@ def get_binding
 
 namespace :android do
   task :ask_stuff do
-    set :output_folder, i.ask('Relative path to output? ')
+    output_folder = fetch(:path, nil)
+    project_name = fetch(:name, nil)
+    sanitized_name = fetch(:sanitized, nil)
+    package = fetch(:package, nil)
 
-    project_name = nil
     loop do
-      project_name = i.ask('Project name? (no trailing -android) ')
-      if project_name.empty?
-        puts "Name is required!"
+      if (output_folder or "").strip.empty?
+        puts "Relative path to output is required!"
+        output_folder = i.ask('Relative path to output? ')
       else
         break
       end
     end
 
-    sanitized_name = project_name.gsub(/[^A-Za-z0-9]/, '')
-    package = "mobi.lab.#{sanitized_name.downcase}"
-
-    ok = false
-    while not ok
-      puts "You have chosen the name #{project_name}"
-      puts "Current names are:\n"
-      puts "package:     #{package}"
-      puts "sanitized:   #{sanitized_name}"
-      puts "name:        #{project_name}"
-
-      i.choose do |menu|
-        menu.prompt = 'This looks ok to you?  '
-        menu.choice(:yes) do
-          puts 'Cool.'
-          ok = true
-        end
-        menu.choice(:no) do
-          package = i.ask('Package: ') { |q| q.default = package }
-          sanitized_name = i.ask('Sanitized name: ') { |q| q.default = sanitized_name }
-          project_name = i.ask('Project name: ') { |q| q.default = project_name }
-        end
+    loop do
+      if (project_name or "").strip.empty?
+        puts "Project name is required!"
+        project_name = i.ask('Project name? (no trailing -android) ')
+      else
+        break
       end
     end
 
+    if (sanitized_name or "").strip.empty?
+      sanitized_name = project_name.gsub(/[^A-Za-z0-9]/, '')
+    end
+
+    if (package or "").strip.empty?
+      puts "Using default package name"
+      package = "mobi.lab.#{sanitized_name.downcase}"
+    end
+
+    puts "You have chosen the project name #{project_name}"
+    puts "Output folder is: #{output_folder}"
+    puts "Names are:\n"
+    puts "package:     #{package}"
+    puts "sanitized:   #{sanitized_name}"
+    puts "name:        #{project_name}"
+
```

Comment:
The old while loop + menu implementation should be used here. At the moment, the values are printed out and the script continues. But it should show the chosen values (either passed in from CLI or entered by the user) and allow the user to confirm or change values like before.

Only the initial input part needs to be changed to either ask for a value or take if from passed in arguments

Reply:
But in what case should the confirmation menu be shown? When the user doesn't pass any arguments from the CLI? Because we cannot have the confirmation in the CI as we cannot pass user input if I understand correctly?

Reply:
Works better now. Can verify that the values are correct before final confirmation.